### ISSUE TITLE
Guild / Team / Clan System

### ISSUE THREAD

**Issue by GTink911** [zorkind/Hellion-Rescue-Project#44]
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
A guild system so players can team up. Automatically adds all guild members to any CM owned by any member of the group.

**Describe alternatives you've considered**
None.

**Additional context**
None.

**Comment by CheeseJedi** [comment]
@zorkind While developing Hellion Explorer I did some work on bringing Steam Group membership to crew lists of 'tagged' vessels in game (the CrewSync console app). The idea was to match a 'clantag' prefix in vessel names and modify the crew list membership so all members of that steam group can access 'faction property' such as vessels and stations. It didn't remove existing members so ships could have additional 'custom' permissions on them. I tested it on a save file from the Hellion Survivors Alliance server (30-40 players on it at the time) and it seemed to work pretty well.

My implementation was a console app that ran against an offline save file (required server downtime) however I suspect this could be more beneficial if it's done in-game somehow. 

A link to the CrewSync program in the Hellion Explorer repo:
https://github.com/CheeseJedi/HELLION.Explorer/blob/master/HELLION.CrewSync/CrewSyncProgram.cs

Perhaps this could give you a shove in the right direction (if this is the right sort of direction at all for this project!) or at least maybe give you some ideas as a starting point?

**Comment by CheeseJedi** [comment]
I've attached an example log file from the CrewSync run on the HSA server (27 players at the time) as the output from the app should be pretty self-explanatory as to what's going on, and might shed some light as to whether this is the right sort of approach for HRP.

[20180625T121335_HELLION.CrewSync.log](https://github.com/zorkind/Hellion-Rescue-Project/files/5115537/20180625T121335_HELLION.CrewSync.log)

**Comment by zorkind** [comment]
Sounds good, altho it is interesting to make use of steam user groups with the "clan/guild" system, i am not sure it is interesting to make it mandatory, but it's something to think about when we get to work on this one. 👍

**Comment by CheeseJedi** [comment]
Agreed on not making it mandatory - perhaps some sort of in-game 'group' that can either contain in-game characters directly or optionally connected to a Steam Group to dictate membership would give the most flexibility to players.
